modules/ROOT/pages/_partials/commons/common-new-logging-api.adoc

```diff
@@ -0,0 +1,634 @@
+`setMaxKeptFiles` is the maximum number of log files that will exist at any time and is the count of rotated files (setRotateCount) plus one.
+
+
+
+ifdef::is-android[]
+
+[tabs]
+=====
+Java::
++
+--
+
+.Old API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=new-file-logging]
+
+----
+--
+
+Kotlin::
++
+--
+
+.Old API
+:!example-caption:
+[source, kotlin, indent=0]
+----
+
+include::android:example$kotlin_snippets/app/src/main/kotlin/com/couchbase/codesnippets/BasicExamples.kt[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, kotlin, indent=0]
+----
+
+include::android:example$kotlin_snippets/app/src/main/kotlin/com/couchbase/codesnippets/BasicExamples.kt[tags=new-file-logging]
+
+----
+--
+=====
+
+endif::is-android[]
+ifdef::is-java[]
+
+.Old API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=new-file-logging]
+
+----
+
+endif::is-java[]
+ifdef::is-csharp[]
+
+.Old API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=file-logging]
+----
+
+.New API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=new-file-logging]
+----
+endif::is-csharp[]
+ifdef::is-objc[]
+
+.Old API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=new-file-logging]
+
+----
+endif::is-objc[]
+ifdef::is-swift[]
+
+.Old API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=new-file-logging]
+
+----
+endif::is-swift[]
+ifdef::is-c[]
+
+.Old API
+:!example-caption:
+[source, c, indent=0]
+----
+
+include::c:example$code_snippets/main.cpp[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, c, indent=0]
+----
+
+include::c:example$code_snippets/main.cpp[tags=new-file-logging]
+
+----
+endif::is-c[]
+
+[#lbl-custom-logsink]
+=== Using a Custom Logger
+
+Installing a custom log sink with the new API is also streamlined: create an
+instance of your custom sink and use `LogSinks.get().setCustom` to install it.
```

Comment:
Maybe `Use LogSinks.get().setCustom in Java or the corresponding static methods on other platforms.`

Reply:
Done